### ISSUE TITLE
Can't take programmatic screen shot on some views - SwiftUI

### ISSUE THREAD

**Issue by ttrompeter** [gualtierofrigerio/ScreenshotSwiftUI#2]
I am working on a SwiftUI app for iPad using Xcode 13.4.1 and iOS Deployment target 15.5 on an Apple Mini running macOS Monterey  version 12.5.1.

On one view I can successfully take a screenshot using your code both programmatically (in .onAppear {} for the view) and also by using a button. On an Action Sheet that is called from that view I can't take the screenshot programmatically but can take it using a button. This same Action Sheet is called in another part of th app and the same situation occurs - can take a screenshot with the button but not programmatically. The same problem is happening on another view were I can't take the screenshot programmatically but can take it using a button. I am unable to discern any difference in my code on the three views so I don't know why it works programmatically in one view and not in the other views or the Action Sheet; yet works with a button in all three.

My active files where I want to take the programmatic screenshots are the MatchView (which works) and the Action Sheet displaying the StatisticsAdminView (which doesn't work for programmatic but does work with a button). An extract of the MatchView is attached since the complete file is huge. The entire StatisticsAdminView is attached.

For testing and trying to figure out the problem I also tried screenshots in the BottomButtonsView and the FunctionalityButtonsView files. The programmatic snapshot does not work in either one, but the snapshot using a button does work in the BottomButtonsView. These two files are also attached.

Any thoughts, ideas or suggestions? Any help would be greatly appreciated. I have been troubleshooting this for major parts of 3 days and am getting nowhere.

I am using your improved recommended code (Which is great - Thank You)

[MatchViewExtract.pdf](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/files/9561444/MatchViewExtract.pdf)
[StatisticsAdminView.pdf](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/files/9561468/StatisticsAdminView.pdf)
[FunctionalityButtonsView.pdf](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/files/9561471/FunctionalityButtonsView.pdf)
[BottomButtonsView.pdf](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/files/9561474/BottomButtonsView.pdf)

**Comment by gualtierofrigerio** [comment]
I cannot read all the files now so I'm asking you a question: the problem is only when you try to take the screenshot inside .onAppear?

**Comment by ttrompeter** [comment]
Yes. The problem only happens inside onAppear. Yeet it also works in the one view also inside onAppear.

**Comment by ttrompeter** [comment]
I switched over the Package version from your Git repository and it now seems to work fine. I am not using onAppear but using the .screenshotMaker { ... } directly on a VStack. I am testing further and will give you an update when I have completed more extensive testing.

Thank you for taking the time to look at my issue. That is above and beyond what is expected.

**Comment by gualtierofrigerio** [comment]
Great :)
Did you come across this repo by reading my blog post? https://www.gfrigerio.com/take-screenshots-of-swiftui-views/
If you have doubts or feedbacks about my article, let me know

**Comment by ttrompeter** [comment]
Yes, I found your information by reading your blog post. I have no comments to your post. It is pretty clear.

I completed testing in my app using the package version of your code and everything works. I still don't know why my earlier code didn't work, but I am moving on. Again, thank you for this screenshot implementation. Very impressive.
